assets/js/components/SelectionBox.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * SelectionBox component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Checkbox } from 'googlesitekit-components';
+
+export default function SelectionBox( {
+	checked,
+	children,
+	disabled,
+	id,
+	onChange,
+	title,
+	value,
+} ) {
+	return (
+		<div
+			className={ classnames( 'googlesitekit-selection-box', {
+				'googlesitekit-selection-box--disabled': disabled,
+			} ) }
+			tabIndex={ -1 }
+			role="button"
```

Comment:
Added `button` role to retain and show focus.

Reply:
Hi @nfmohit, I can see you've pushed an update for this. I do humbly apologise, but seeing the required code and also reviewing the behaviour, I'm actually not so sure it's the right course to take.

Code-wise, it looks like you've had to add a keyboard event handler to satisfy the `jsx-a11y/click-events-have-key-events` ESLint rule, but from my reading, it's not really necessary to handle key events here to provide the given functionality (we can't tab onto the element, so the only way to realistically get it to receive keyboard events is to click to focus it, in which case the checkbox will become focused anyway). So this starts to add a bit of unnecessary complexity - of course we could simply disable the `jsx-a11y/click-events-have-key-events` rule for this line.

However, a second point is the use of the `button` role... Pressing a button should "do" something, now, arguably this does do something because it focuses the checkbox and highlights the selection box, but it's a bit minimalist as far as buttons go, and it might be a bit unintuitive from an accessibility perspective.

This brings me to my main point, which is simply that the UX feels a little confusing - clicking on the selection box looks visually similar to clicking on the checkbox, but without actually toggling it. It could become confusing - I can imagine clicking just outside the checkbox, maybe not paying it my full attention, and then being unsure whether I've just toggled something or not. I think this can introduce a sense of cognitive dissonance where we can simply avoid it by _not_ allowing the selection box to be focused by clicking on it, and leave the interaction here purely to the checkbox, which is really the only control that actually makes a tangible difference and modifies the state.

Like I said, I am sorry for providing this direction in the first instance as I can see you've spent a bit of time on it, but I think we'd be better off simply removing these clicking handlers and the `button` role. Please let me know what you think - if you like we could ask Sigal for her opinion for a UX perspective on this one.

Reply:
Hi @techanvil!
Thank you for your kind feedback on this, and no worries at all!

As discussed, I have removed the button behaviour from this `div` overall. I have also asked Sigal [internally](https://10up.slack.com/archives/C01CA828ZPZ/p1696752549658799) to see if we expect any different UX behaviour here. If so, we can probably address that in a follow-up issue (we'll probably need one anyway to increase the width of the panel and relocate some elements between the header and footer).

Thanks!